python/cudf/cudf/utils/gpu_utils.py

```diff
@@ -0,0 +1,110 @@
+def validate_setup():
+    from cudf._cuda.gpu import (
+        getDeviceCount,
+        driverGetVersion,
+        runtimeGetVersion,
+        getDeviceAttribute,
+        CudaDeviceAttr,
+        getDeviceProperties,
+        CUDARuntimeError,
+    )
+    import warnings
+
+    try:
+        gpus_count = getDeviceCount()
+    except CUDARuntimeError:
+        # If there is no GPU detected, set `gpus_count` to -1
+        gpus_count = -1
+
+    if gpus_count > 0:
+        # Cupy throws RunTimeException to get GPU count,
+        # hence obtaining GPU count by in-house cpp api above
+
+        # 75 - Indicates to get "cudaDevAttrComputeCapabilityMajor" attribute
+        # 0 - Get GPU 0
+        major_version = getDeviceAttribute(
+            CudaDeviceAttr.cudaDevAttrComputeCapabilityMajor, 0
+        )
+
+        if major_version >= 6:
+            # You have a GPU with NVIDIA Pascal™ architecture or better
+            # Hardware Generation	Compute Capability
+            #    Turing	                7.5
+            #    Volta	                7.x
+            #    Pascal	                6.x
+            #    Maxwell	              5.x
+            #    Kepler	                3.x
+            #    Fermi	                2.x
+            pass
+        else:
+            device_props = getDeviceProperties(0)
```

Comment:
This is an expensive call, is there anyway we can get just the name?

See: https://devblogs.nvidia.com/cuda-pro-tip-the-fast-way-to-query-device-properties/

Reply:
I think we can use `cuDeviceGetName` : https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__DEVICE.html#group__CUDA__DEVICE_1gef75aa30df95446a845f2a7b9fffbb7f

Reply:
Added `cuDeviceGetName` cython api and fetching from it. Removed the `getDeviceProperties` call but left the cython API intact for any future use.